std/js/_std/Type.hx

```diff
@@ -157,6 +157,30 @@ enum ValueType {
 		return createEnum(e,c,params);
 	}
 
+	#if (js_es >= 6)
+	public static function getInstanceFields( c : Class<Dynamic> ) : Array<String> {
+		var fields:Null<Array<String>> = (cast c).__instanceFields__;
+		return if (fields == null) [] else fields.copy();
+	}
+
+	public static function getClassFields( c : Class<Dynamic> ) : Array<String> {
+		var a = js.Object.getOwnPropertyNames(cast c);
+		a.remove("__id__");
+		a.remove("hx__closures__");
+		a.remove("__name__");
+		a.remove("__interfaces__");
+		a.remove("__properties__");
+		a.remove("__instanceFields__");
+		a.remove("__super__");
+		a.remove("__meta__");
+		a.remove("prototype");
+		a.remove("name");
```

Comment:
Why do `name` and `length` get removed?

Reply:
Because those are reserved fields of js `Function` (name and arg count). If Haxe's class have statics `name` and `length`, they are escaped by `$` in the generated code.

Reply:
They weren't present in `Reflect.fields` result probably because they are not enumerable.

Reply:
But what if my class actually has `name` or `length` field?

Reply:
As said, they will get escaped:
```haxe
class Main {
	static var name = "hi";
	static var length = 3;
}
```
```js
class Main {
}
Main.$name = "hi";
Main.$length = 3;
```

this is already the behaviour for genjs.